### ISSUE TITLE
CSS refresh not considering removed properties on live reload

### ISSUE THREAD

**Issue by kutoman** [codenameone/CodenameOne#2698]
When I for example have this:
```
MyStlye {
  color: blue;
}
```

and remove a specific property:
```
MyStlye {
}
```

the `Component` with `MyStyle` still keeps the (removed) style property in the simulation

**Comment by shannah** [comment]
I can't seem to reproduce this.  I have added a sample in the [SamplesRunner](https://github.com/codenameone/CodenameOne/tree/master/Samples) to try to reproduce.

To run the sample, check out the CodenameOne sources from Github, and run the samples target:

~~~
$ git clone https://github.com/codenameone/CodenameOne
$ cd CodenameOne
$ ant samples
~~~~

That will open the Codename One Samples runner, which looks like:

![image](https://user-images.githubusercontent.com/2677562/55020911-0ef31f80-4fb5-11e9-9286-18efec58afb8.png)

In the "HelloWorldCSS" row, press the "Launch" button to run the sample in the simulator.

![image](https://user-images.githubusercontent.com/2677562/55020995-3e099100-4fb5-11e9-85ff-462d49bd14b3.png)

**Editing the CSS File**

With or without the sample running, you can edit the CSS file by going to "More..." > "Edit CSS File"

![image](https://user-images.githubusercontent.com/2677562/55021070-60031380-4fb5-11e9-818f-5b609ed3028b.png)

This will open the CSS file for editing.  It will look like:

~~~~
#Constants {
    includeNativeBool: true; 
}

Label {
    color:red;
}

MyStyle {
    
}
~~~~

You can change the MyStyle style to:

~~~~
MyStyle {
    color:blue;
}
~~~~

And save the file.

Then, with the sample still running, select "More..." > "Refresh CSS"

![image](https://user-images.githubusercontent.com/2677562/55021197-92ad0c00-4fb5-11e9-8e90-e803c77eceb6.png)

After a few seconds you should see the text in the sample turn blue.

Now go back to the CSS file, and remove the "color:blue" that you added.  Then select "More..." > "Refresh CSS" again.  After a few seconds, the text will change back to black.

Please let me know if you get different results, and also please try to modify this sample to reproduce your issue.

**Comment by kutoman** [comment]
yeah, in that sample it works properly, but is this the same refresh mechanism as the **auto** refresh during a simulator run? It still doesn't work in my samples.

**Comment by shannah** [comment]
It uses the same auto-refresh mechanism.  IDE shouldn't affect this either, as the simulator itself is monitoring the CSS file for changes, and then recompiling as necessary.

**Comment by kutoman** [comment]
hmm ok, I'll come back if I spot  any new hint. Maybe other parts of the css code are somehow blocking the complete refresh since I'm facing the issue in an extensive css.
